### ISSUE TITLE
Limit net imports

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/calliope#270]
## Problem description

I would like to limit the net imports of a certain carrier into a location or group of locations over the entire time horizon of my model. I thought I could do that with Calliope 0.6.4 in the following way: add a demand_share_min constraint to all internal supply technologies, so that they generate the expected share.

While this works, it's not the same as limiting the net imports: actual carrier consumption is higher within the locations because there are losses in storage and transmission. Using the method described above, those losses can be fed with net imports from the grid. Thus, there are in fact higher net imports in the end. In a case in which I disallow all net imports, I actually see 5% net imports.

Instead one should probably model net import limitations with a demand_share_max constraint on the transmission links. Transmission links cannot be part of group constraints atm though.

## Steps to reproduce the problem

```yaml
group_constraints:
    import_restriction_0_percent:
        locs: ["1"]
        techs: ["ac_transmission"] # this does not work
        demand_share_max:
            electricity: 0.0
```

## Calliope version

0.6.4

**Comment by brynpickering** [comment]
Yup, it would be good to include transmission technologies in the group constraints. They do, however, need a lot of special treatment. You could have transmission be associated with a new carrier (e.g. AC_transmission) and have a conversion tech to convert that to demanded electricity at each location. Then you could place the group constraint on the conversion tech?

**Comment by timtroendle** [comment]
Thanks a lot for the idea. Sounds like a pragmatic workaround. I'm only afraid it would make result analysis slightly more complicated (I very much got used to `squeeze("carriers")` 😉) and much more importantly, having two carriers and having conversion technology may make the problem more complicated to solve. Certainly it's worth a try.

**Comment by brynpickering** [comment]
Well, you now just need to get used to using `sum("carriers")` 😉. I would be certainly interested to hear what the slow-down is. I think the commercial solvers are quite good at preprocessing these kinds of implementations to the same sized optimisation problem.

**Comment by timtroendle** [comment]
I'll let you know.

**Comment by timtroendle** [comment]
Actually, I fear this doesn't work either without changes in Calliope. It works well for restricting _gross_ imports, but not for _net_ imports. At least I do not see a way of restricting net imports. I can restrict the `carrier_prod` of any supply tech, but I would need to restrict the sum of `carrier_prod` plus `carrier_con` (either of the same tech if possible, or of two techs).

Perhaps this could be solved by optionally choosing demand techs (already discussed in #227):

```yaml
techs:
    to_grid:
        essentials:
            name: 'Export electricity to transmission grid'
            parent: conversion
            carrier_in: electricity
            carrier_out: electricity_in_transmission
    from_grid:
        essentials:
            name: 'Import electricity from transmission grid'
            parent: conversion
            carrier_in: electricity_in_transmission
            carrier_out: electricity
group_constraints.import_restriction_0_percent:
            locs: [some_location]
            techs: [from_grid]
            demand_share_max:
                electricity: 0.0
                demand_techs: [electricity_demand, to_grid]
```

**Comment by sjpfenninger** [comment]
Addressed in #271
